runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -201,6 +213,21 @@ private XWalkCoreWrapper(Context context, int minApiVersion) {
         mWrapperContext = context;
     }
 
+    private boolean isDownloadMode() {
+        try {
+            PackageManager packageManager = mWrapperContext.getPackageManager();
+            ApplicationInfo appInfo = packageManager.getApplicationInfo(
+                    mWrapperContext.getPackageName(), PackageManager.GET_META_DATA);
```

Comment:
For shared mode, the developer doesn't need to specify it specially, just link to the shared library. For download mode, it should be the same, we don't want to force the developer to do extra settings.